h2o-automl/src/main/java/ai/h2o/automl/targetencoding/TargetEncoderModel.java

```diff
@@ -0,0 +1,158 @@
+package ai.h2o.automl.targetencoding;
+
+import hex.Model;
+import hex.ModelCategory;
+import hex.ModelMetrics;
+import hex.ModelMojoWriter;
+import water.H2O;
+import water.Iced;
+import water.Key;
+import water.MRTask;
+import water.fvec.Chunk;
+import water.fvec.Frame;
+import water.util.IcedHashMap;
+
+import java.util.HashMap;
+import java.util.Map;
+
+
+public class TargetEncoderModel extends Model<TargetEncoderModel, TargetEncoderModel.TargetEncoderParameters, TargetEncoderModel.TargetEncoderOutput> {
+
+  public TargetEncoderModel(Key<TargetEncoderModel> selfKey, TargetEncoderParameters parms, TargetEncoderOutput output) {
+    super(selfKey, parms, output);
+  }
+
+  @Override
+  public ModelMetrics.MetricBuilder makeMetricBuilder(String[] domain) {
+    throw H2O.unimpl("No Model Metrics for TargetEncoder.");
+  }
+
+  @Override
+  protected double[] score0(double[] data, double[] preds) {
+    return new double[0];
+  }
+
+  private static IcedHashMap<String, Map<String, TEComponents>> _targetEncodingMap;
+
+
+  public static class TargetEncoderParameters extends Model.Parameters {
+    @Override
+    public String algoName() {
+      return "TargetEncoder";
+    }
+
+    @Override
+    public String fullName() {
+      return "TargetEncoder";
+    }
+
+    @Override
+    public String javaName() {
+      return TargetEncoderModel.class.getName();
+    }
+
+    @Override
+    public long progressUnits() {
+      return 0;
+    }
+
+
+    public void addTargetEncodingMap(Map<String, Frame> encodingMap) {
+      IcedHashMap<String, Map<String, TEComponents>> transformedEncodingMap = new IcedHashMap<>();
+      Map<String, FrameToTETable> tasks = new HashMap<>();
+
+      for (Map.Entry<String, Frame> entry : encodingMap.entrySet()) {
+
+        Frame encodingsForParticularColumn = entry.getValue();
+        FrameToTETable task = new FrameToTETable().dfork(encodingsForParticularColumn);
+
+        tasks.put(entry.getKey(), task);
+      }
+
+      for (Map.Entry<String, FrameToTETable> taskEntry : tasks.entrySet()) {
+        transformedEncodingMap.put(taskEntry.getKey(), taskEntry.getValue().getResult().table);
+      }
+      _targetEncodingMap = transformedEncodingMap;
+    }
+    
+  }
+
+  public static class TargetEncoderOutput extends Model.Output {
+    public TargetEncoderOutput(TargetEncoderBuilder b) { super(b); }
+
+    public transient Map<String, Map<String, int[]>> _target_encoding_map = convertEncodingMap(_targetEncodingMap); 
+
+
+    @Override public ModelCategory getModelCategory() {
+      return ModelCategory.TargetEncoder;
+    }
+  }
+
+
+  public static Map<String, Map<String, int[]>> convertEncodingMap(IcedHashMap<String, Map<String, TEComponents>> em) {
```

Comment:
It would be nice if TE map was represented by a dedicated class instead of `Map<String, Map<String, int[]>>`, `IcedHashMap<String, Map<String, TEComponents>>`. It would improve code readability.